Source/WebKit/UIProcess/Inspector/mac/WebInspectorUIProxyMac.mm

```diff
@@ -284,6 +381,57 @@ - (void)inspectorViewController:(WKInspectorViewController *)inspectorViewContro
     return window;
 }
 
+void WebInspectorUIProxy::showSavePanel(NSWindow *frontendWindow, NSURL *platformURL, Vector<InspectorFrontendClient::SaveData>&& saveDatas, bool forceSaveAs, CompletionHandler<void(NSURL *)>&& completionHandler)
+{
+    ASSERT(platformURL);
+
+    RetainPtr savePanel = [NSSavePanel savePanel];
+    [savePanel setExtensionHidden:NO];
+
+    auto controller = adoptNS([[WKWebInspectorUISaveController alloc] initWithSaveDatas:WTFMove(saveDatas) savePanel:savePanel.get()]);
+
+    auto saveToURL = [controller, completionHandler = WTFMove(completionHandler)] (NSURL *actualURL) mutable {
+        ASSERT(actualURL);
+
+        if ([controller base64Encoded]) {
+            auto decodedData = base64Decode([controller content], Base64DecodeOptions::ValidatePadding);
+            if (!decodedData)
+                return;
+            auto dataContent = adoptNS([[NSData alloc] initWithBytes:decodedData->data() length:decodedData->size()]);
+            [dataContent writeToURL:actualURL atomically:YES];
+        } else
+            [[controller content] writeToURL:actualURL atomically:YES encoding:NSUTF8StringEncoding error:NULL];
+
+        completionHandler(actualURL);
+    };
+
+    if (!forceSaveAs) {
+        saveToURL(platformURL);
+        return;
+    }
+
+    [savePanel setNameFieldStringValue:platformURL.lastPathComponent];
```

Comment:
Should the name also be changed when the `Format` is changed? Seems kinda odd to pipe individual suggested names all the way to the UI process to only end up using one of them...

Reply:
unfortunately it's not possible to change the name after a `NSSavePanel` is shown

changing the `allowedContentTypes` will adjust the extension though (but keep whatever the user typed for the filename as is)